fil_finder/utilities.py

```diff
@@ -140,7 +140,8 @@ def distance(x, x1, y, y1):
 
 def padwithzeros(vector, pad_width, iaxis, kwargs):
     vector[:pad_width[0]] = 0
-    vector[-pad_width[1]:] = 0
+    if pad_width[1] > 0:
+        vector[-pad_width[1]:] = 0
```

Comment:
Yes, but as you say, it shouldn't be possible for `pad_width[0] != pad_width[1]`.  You just can't allow 0 here, because `vector[0:] == vector[:] != vector[:0]`